src/doc/src/guide/build-cache.md

```diff
@@ -1,108 +1 @@
-<code style="white-space: nowrap">target/doc/</code> | Contains rustdoc documentation ([`cargo doc`]).
-<code style="white-space: nowrap">target/package/</code> | Contains the output of the [`cargo package`] and [`cargo publish`] commands.
-
-Cargo also creates several other directories and files needed for the build
-process. Their layout is considered internal to Cargo, and is subject to
-change. Some of these directories are:
-
-Directory | Description
-----------|------------
-<code style="white-space: nowrap">target/debug/deps/</code> | Dependencies and other artifacts.
-<code style="white-space: nowrap">target/debug/incremental/</code> | `rustc` [incremental output], a cache used to speed up subsequent builds.
-<code style="white-space: nowrap">target/debug/build/</code> | Output from [build scripts].
-
-## Dep-info files
-
-Next to each compiled artifact is a file called a "dep info" file with a `.d`
-suffix. This file is a Makefile-like syntax that indicates all of the file
-dependencies required to rebuild the artifact. These are intended to be used
-with external build systems so that they can detect if Cargo needs to be
-re-executed. The paths in the file are absolute by default. See the
-[`build.dep-info-basedir`] config option to use relative paths.
-
-```Makefile
-# Example dep-info file found in target/debug/foo.d
-/path/to/myproj/target/debug/foo: /path/to/myproj/src/lib.rs /path/to/myproj/src/main.rs
-```
-
-## Shared cache
-
-A third party tool, [sccache], can be used to share built dependencies across
-different workspaces.
-
-To setup `sccache`, install it with `cargo install sccache` and set
-`RUSTC_WRAPPER` environmental variable to `sccache` before invoking Cargo. If
-you use bash, it makes sense to add `export RUSTC_WRAPPER=sccache` to
-`.bashrc`. Alternatively, you can set [`build.rustc-wrapper`] in the [Cargo
-configuration][config]. Refer to sccache documentation for more details.
-
-[`RUSTFLAGS`]: ../reference/config.md#buildrustflags
-[`build.dep-info-basedir`]: ../reference/config.md#builddep-info-basedir
-[`build.rustc-wrapper`]: ../reference/config.md#buildrustc-wrapper
-[`build.target-dir`]: ../reference/config.md#buildtarget-dir
-[`cargo doc`]: ../commands/cargo-doc.md
-[`cargo package`]: ../commands/cargo-package.md
-[`cargo publish`]: ../commands/cargo-publish.md
-[build scripts]: ../reference/build-scripts.md
-[config]: ../reference/config.md
-[def-workspace]:  ../appendix/glossary.md#workspace  '"workspace" (glossary entry)'
-[target]: ../appendix/glossary.md#target '"target" (glossary entry)'
-[environment variable]: ../reference/environment-variables.md
-[incremental output]: ../reference/profiles.md#incremental
-[sccache]: https://github.com/mozilla/sccache
-[profile]: ../reference/profiles.md
-[binary executables]: ../reference/cargo-targets.md#binaries
-[library targets]: ../reference/cargo-targets.md#library
-[example targets]: ../reference/cargo-targets.md#examples
+# Build Cache
```

Comment:
Looks like there are some more outdated links.

```
src/doc/src/reference/cargo-targets.md
371:[Build cache]: ../guide/build-cache.md

src/doc/src/reference/config.md
557:info](../guide/build-cache.md#dep-info-files) file paths. This config setting

src/doc/src/reference/workspaces.md
293:[output directory]: ../guide/build-cache.md

src/doc/src/reference/profiles.md
374:[`target` directory]: ../guide/build-cache.md

src/doc/book.toml
12:"/guide/build-cache.html" = "../reference/build-cache.html"

src/doc/man/includes/options-target-triple.md
15:[build cache](../guide/build-cache.html) documentation for more details.
```

And the one in `src/doc/src/reference/config.md` will be an invalid link. 

I think there are a few options

* Just fix the wrong links above, and merge this PR.
* Move the chapter without renaming to `reference/guide`.
* Put a `<script>` redirection for hash anchors in this empty chapter, instead of using `[output.html.redirect]`.

Reply:
Yeah, the link checker isn't able to cache these. I've fixed them.

As for the anchor, I thought we weren't too concerned about that for this case?

Reply:
Looked it again. It is an intra-link inside The Cargo Book so should always be valid I believe. And according to [my test](https://github.com/rust-lang/cargo/pull/14620#discussion_r1786690819) against rust-lang/rust it doesn't seem to cause any link check problem.